Stop OmegaConfigLoader from reading config from hidden directory like `ipynb_checkpoints` (#2977)
kedro-org/kedro: RELEASE.md

* Updated `kedro pipeline create` and `kedro catalog create` to use new `/conf` file structure.
* Converted `setup.py` in default template to `pyproject.toml` and moved flake8 configuration
  to dedicated file `.flake8`.

> * Fixed a bug that `OmegaConfigLoader` read config from hidden directory like `.ipynb_checkpoints`.

Reviewer: This wasn't removed btw

Author: Thanks, I missed this when I merged from main and get conflicts

* Updated `kedro pipeline create` and `kedro catalog create` to use new `/conf` file structure.
* Converted `setup.py` in default template to `pyproject.toml` and moved flake8 configuration
  to dedicated file `.flake8`.

> * Fixed a bug that `OmegaConfigLoader` read config from hidden directory like `.ipynb_checkpoints`.

Reviewer: ```suggestion
* Updated `OmegaConfigLoader` to ignore config from hidden directories like `.ipynb_checkpoints`.
```
